OutOfSchool/OutOfSchool.WebApi.Tests/Hubs/ChatWorkshopHubTests.cs

```diff
@@ -0,0 +1,262 @@
+                .Returns(Task.CompletedTask);
+
+            // Act
+            await chatHub.OnConnectedAsync();
+
+            // Assert
+            groupsMock.Verify(x => x.AddToGroupAsync(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<CancellationToken>()), Times.Exactly(validChatRoomIds.Count));
+        }
+
+        [Test]
+        public void OnConnectedAsync_IfOneOfTheClaimsIsNotFoundInJWT_ThrowsAuthenticationException()
+        {
+            // Arrange
+            var userRole = Role.Provider.ToString();
+            hubCallerContextMock.Setup(x => x.User.FindFirst("role"))
+                .Returns(default(Claim));
+
+            // Act and Assert
+            Assert.ThrowsAsync<AuthenticationException>(async () => await chatHub.OnConnectedAsync());
+        }
+
+        [Test]
+        public async Task SendMessageToOthersInGroup_IfOneOfTheClaimsIsNotFoundInJWT_ShouldWriteMessageToCallerWithException()
+        {
+            // Arrange
+            var userRole = Role.Provider.ToString();
+            hubCallerContextMock.Setup(x => x.User.FindFirst("role"))
+                .Returns(new Claim(ClaimTypes.NameIdentifier, userRole));
+            hubCallerContextMock.Setup(x => x.User.FindFirst("sub"))
+                .Returns(default(Claim));
+
+            var chatNewMessage = "{'workshopId':1, 'parentId':1, 'text':'hi', 'senderRoleIsProvider':false }";
+            clientsMock.Setup(clients => clients.Caller).Returns(clientProxyMock.Object);
+
+            // Act
+            await chatHub.SendMessageToOthersInGroupAsync(chatNewMessage).ConfigureAwait(false);
+
+            // Assert
+            clientsMock.Verify(clients => clients.Caller, Times.Once);
+            clientsMock.Verify(clients => clients.OthersInGroup(It.IsAny<string>()), Times.Never);
+        }
+
+        [Test]
+        public async Task SendMessageToOthersInGroup_WhenStringIsInvalid_ShouldWriteMessageToCallerWithException()
+        {
+            // Arrange
+            var chatNewMessage = "string with wrong format";
+            clientsMock.Setup(clients => clients.Caller).Returns(clientProxyMock.Object);
+
+            // Act
+            await chatHub.SendMessageToOthersInGroupAsync(chatNewMessage).ConfigureAwait(false);
+
+            // Assert
+            clientsMock.Verify(clients => clients.Caller, Times.Once);
+            clientsMock.Verify(clients => clients.OthersInGroup(It.IsAny<string>()), Times.Never);
+        }
+
+        [Test]
+        public async Task SendMessageToOthersInGroup_WhenUserSetsNotOwnParentId_ShouldWriteMessageToCallerWithException()
+        {
+            // Arrange
+            var userRole = Role.Parent.ToString();
+            hubCallerContextMock.Setup(x => x.User.FindFirst("role"))
+                .Returns(new Claim(ClaimTypes.NameIdentifier, userRole));
+
+            var invalidParentId = 1;
+
+            var chatNewMessage = "{'workshopId':1, 'parentId':1, 'text':'hi', 'senderRoleIsProvider':false }";
```

Comment:
`'parentId':1` should be `'parentId':" + invalidParentId.ToString() + "`, I guess